### PR TITLE
[SMALLFIX] Fix broken sync directory retry

### PR DESCRIPTION
**Pull request by bf8086**
Fix infinite retries when a directory cannot be created in UFS due to permission issue.

### CODE REVIEW

**Comment by bf8086** [comment]
@aaudiber @gpang PTAL.

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java [hunk 1]
```diff
@@ -1021,10 +1020,9 @@ public void syncPersistDirectory(InodeDirectory dir, JournalContext journalConte
               dir.setOwner(status.getOwner())
                   .setGroup(status.getGroup())
                   .setMode(status.getMode());
-            } catch (FileNotFoundException e) {
-              // Retry creation given that the directory might have just been removed.
-              LOG.warn("Directory {} no longer exists on UFS. Retry creation.", ufsUri);
-              continue;
+            } catch (Exception e) {
+              throw new IOException(String.format("Cannot create or sync UFS directory %s.",
```
Comment (aaudiber) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java]: include the message of the root cause in the detail string - only the detail string is sent over the wire, not the cause chain

Comment (aaudiber) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java]: can we specify whether the issue was with creation vs with sync?

Comment (bf8086) [reply core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java]: Done.

**Pull request merged**